pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -168,6 +207,160 @@ func newLighthouseAgent(cr *submarinerv1alpha1.ServiceDiscovery) *appsv1.Deploym
+			},
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            lighthouseCoreDNSName,
+							Image:           getImagePath(cr, lighthouseCoreDNSImage),
+							ImagePullPolicy: "IfNotPresent",
+							Args: []string{
+								"-conf",
+								"/etc/coredns/Corefile",
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{Name: "config-volume", MountPath: "/etc/coredns", ReadOnly: true},
+							},
+						},
+					},
+
+					ServiceAccountName:            "submariner-operator",
+					TerminationGracePeriodSeconds: &terminationGracePeriodSeconds,
+					Volumes: []corev1.Volume{
+						{Name: "config-volume", VolumeSource: corev1.VolumeSource{ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{Name: lighthouseCoreDNSName},
+							Items: []corev1.KeyToPath{
+								{Key: "Corefile", Path: "Corefile"},
+							},
+							DefaultMode: &defaultMode,
+						}}},
+					},
+				},
+			},
+		},
+	}
+}
+
+func newLigthhouseCoreDNSService(cr *submarinerv1alpha1.ServiceDiscovery) *corev1.Service {
+	labels := map[string]string{
+		"app":       lighthouseCoreDNSName,
+		"component": componentName,
+	}
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      lighthouseCoreDNSName,
+			Labels:    labels,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{
+				Name:     "udp",
+				Protocol: "UDP",
+				Port:     53,
+				TargetPort: intstr.IntOrString{Type: intstr.Int,
+					IntVal: 53},
+			}},
+			Type: corev1.ServiceTypeClusterIP,
+			Selector: map[string]string{
+				"app": lighthouseCoreDNSName,
+			},
+		},
+	}
+}
+
+func updateDNSConfigMap(client client.Client, k8sclientSet *clientset.Clientset, cr *submarinerv1alpha1.ServiceDiscovery) error {
+	configMaps := k8sclientSet.CoreV1().ConfigMaps("kube-system")
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		configMap, err := configMaps.Get("coredns", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		/* This entry will be added to config map
+		# lighthouse
+		supercluster.local:53 {
+		    forward . 2.2.2.2:5353
+		}
+		*/
+		corefile := configMap.Data["Corefile"]
+		if strings.Contains(corefile, "lighthouse") {
+			// Assume this means we've already set the ConfigMap up
+			return nil
+		}
+		lighthouseDnsService := &corev1.Service{}
+		err = client.Get(context.TODO(), types.NamespacedName{Name: lighthouseCoreDNSName, Namespace: cr.Namespace,}, lighthouseDnsService)
+		if (err != nil || lighthouseDnsService.Spec.ClusterIP == ""){
+			return stderr.New("lighthouseDnsService ClusterIp should be available")
+		}
+		expectedCorefile := `#lighthouse
+supercluster.local {
+forward . `
+		expectedCorefile = expectedCorefile + lighthouseDnsService.Spec.ClusterIP + "\n" + "}\n"
```

Comment:
We will need to watch the service, if the IP changed for some reason (somebody deletes the service and operator re-creates...) ?

Reply:
Can you open a follow up issue/PR to handle that case? (monitoring the service), also in that case L336 will need to be smarter.

Reply:
ack thanks.

Updating this in the future by the operator will require smarted logic here. Like a regex to modify only the supercluster section.